### DISCUSSION TITLE
[tensorflow-rocm] gfx1031

### DISCUSSION THREAD

**Discussion by shtrophic** [rocm-arch/rocm-arch#828] [Support]
Opening another discussion, as suggested.

So I'm in the process to build `tensorflow-rocm` with my RX 6700XT, which is the gfx1031 platform.
Currently, I'm stuck at building `rocBLAS`, since it fails with the error-message "gfx1031" not supported.

Creating a patch from [this pr](https://github.com/ROCmSoftwarePlatform/rocBLAS/pull/1251) does not suffice, since it does not contain necessary changes for the CMakeFile. I added the latter manually, but the next point of failure is tensile; The creator of the rocBLAS-pr also submitted [one](https://github.com/ROCmSoftwarePlatform/Tensile/pull/1511) to Tensile, but I can't seem to apply that patch, since the build of rocBLAS pulls the tensile repo every time? At least, it seems to be in the review process atm.

Also, I have found [this post on reddit](https://www.reddit.com/r/ROCm/comments/uax358/comment/i9l2oqo/?utm_source=share&utm_medium=web2x&context=3), it says to set `export HSA_OVERRIDE_GFX_VERSION=10.3.0`, seems to be a hacky solution?

Can you give me any guidance on this? I'm sure rocBLAS wouldn't be the only issue though, since rocSPARSE won't build as well -- and I'm sure that they are not the only packages having issues. Am I better off waiting for AMD to get back to their abandonware?

**Comment by acxz** [comment]
> but I can't seem to apply that patch, since the build of rocBLAS pulls the tensile repo every time?

You can apply the patch in the PKGBUILD after rocblas pulls in Tensile.

> rocSPARSE won't build

What's the build log?

> better off waiting for AMD

Up to you, if you think you are capable of trying to fix the issues go for it! Otherwise, you may find yourself wasting a good bit of time.

**Reply by tpkessler** [reply]
> You can apply the patch in the PKGBUILD after rocblas pulls in Tensile.

It's not that easy, unfortunately. The build process of tensile is part of the cmake step of rocblas. One can provide a local (and thus patched) version of tensile.

**Comment by tpkessler** [comment]
> Can you give me any guidance on this?

Wait for the next ROCm release. Patching support for gfx1031 in rocblas is very complicated and I don't think that we should force it by manually patching the 5.2.0 release, given that there are PRs for future releases.

> waiting for AMD to get back to their abandonware?

That's very harsh and I do not agree. As explained by AMD several times, the support (and testing) of newer platforms takes a while. And as you can see in the replies of AMD's employees in the linked PRs (and every else) they really want to have this in ROCm soon.
